### ISSUE TITLE
indent rule issue

### ISSUE THREAD

**Issue by EvgenyOrekhov** [eslint/eslint#7320]
**Tell us about your environment**
- **ESLint Version:** 3.7.1
- **Node Version:** 6.7.0
- **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
{
    "rules": {
        "indent": "error"
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
JSON
    .stringify(
        {
            test: "test"
        }
    );
```

**What did you expect to happen?**
No warnings

**What actually happened? Please include the actual, raw output from ESLint.**
Warnings:
3:9 - Expected indentation of 4 spaces but found 8. (indent)
4:13 - Expected indentation of 8 spaces but found 12. (indent)
5:9 - Expected indentation of 4 spaces but found 8. (indent)

**Comment by not-an-aardvark** [comment]
Thanks for the report. I was able to reproduce this issue in the [demo](http://eslint.org/demo/).

**Comment by sstern6** [comment]
@not-an-aardvark this is related to https://github.com/eslint/eslint/issues/7366 and my solution looks like it fixes this one as well.  Will have a PR with tests up tonight for review.  Looking forward to your feedback.

**Comment by sstern6** [comment]
@not-an-aardvark I believe this can be closed because of https://github.com/eslint/eslint/pull/7400 being merged.  Just confirmed this has correct functionality locally.  Can you please confirm/close?

**Comment by not-an-aardvark** [comment]
Hmm, I can't confirm (maybe I'm doing something wrong).

On latest master:

```js
// foo.js
/* eslint indent: error */
JSON
    .stringify(
        {
            test: "test"
        }
    );
```

```bash
eslint --no-eslintrc foo.js
```

output:

```
  5:9   error  Expected indentation of 4 spaces but found 8   indent
  6:13  error  Expected indentation of 8 spaces but found 12  indent
  7:9   error  Expected indentation of 4 spaces but found 8   indent
```

**Comment by sstern6** [comment]
Hmm, youre right. Will look into this and have a PR open when I can.

**Comment by sstern6** [comment]
@not-an-aardvark still looking into this

With the new CallExpression implemented I want to confirm the eslintrc for this example:

```
{
    "rules": {
      "indent": ["error", 4, {"CallExpression": {"arguments": 4}}]
    }
}
```

This would be the correct config for the above example correct?

Thanks

**Comment by not-an-aardvark** [comment]
No, I think it would be:

```json
{
    "rules": {
      "indent": ["error", 4, {"CallExpression": {"arguments": 1}}]
    }
}
```

because the `1` refers to indent *levels*, which are multiplied by the indent size (4).

**Comment by evilebottnawi** [comment]
@not-an-aardvark the problem remains in the latest version (`3.12.1`), reproduced if you use `chain`
```js
myObject
    .otherFunction()
    .someFunction(
        () => {
            console.log('Here - 1');
        },
        () => {
            console.log('Here - 2');
        }
    );
```
```shell
    4:9   error  Expected indentation of 4 spaces but found 8                           indent
    7:9   error  Expected indentation of 4 spaces but found 8                           indent
```
Config
```json
"rules": {
            "indent": [
                "error",
                4,
                {
                    "SwitchCase": 1,
                    "VariableDeclarator": 1,
                    "outerIIFEBody": 1,
                    "MemberExpression": 1,
                    "FunctionDeclaration": {
                        "body": 1,
                        "parameters": 1
                    },
                    "FunctionExpression": {
                        "body": 1,
                        "parameters": 1
                    },
                    "CallExpression": {
                       "arguments": 1
                    },
                    "ArrayExpression": 1,
                    "ObjectExpression": 1
                }
            ]
        }
```

**Comment by not-an-aardvark** [comment]
@evilebottnawi Thank you for the update. I'm aware that this issue hasn't been fixed yet in a release. (GitHub linked to several commits above that fix the issue, but none of them have actually been merged into master yet. This issue will most likely be fixed when https://github.com/eslint/eslint/pull/7618 is merged.)

**Comment by evilebottnawi** [comment]
@not-an-aardvark Thanks for answer, we will wait merge :smile:
